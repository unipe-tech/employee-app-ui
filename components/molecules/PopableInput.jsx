import EStyleSheet from "react-native-extended-stylesheet";
import React from "react";
import FormInput from "../atoms/FormInput";
import { MaterialIcons } from "react-native-vector-icons";
import { COLORS, FONTS } from "../../constants/Theme";
import { useState } from "react";
import { Popable } from "react-native-popable";

const PopableInput = ({
  containerStyle,
  value,
  onChange,
  placeholder,
  autoCapitalize,
  content,
}) => {
  return (
    <FormInput
      containerStyle={{ ...styles.container, ...containerStyle }}
      placeholder={placeholder}
      value={value}
      autoCapitalize={autoCapitalize}
      onChange={onChange}
      appendComponent={
        <Popable content={content} position="left" caret={false}>
          <MaterialIcons name="info-outline" size={20} color="grey" />
        </Popable>
      }
    />
  );
};

const styles = EStyleSheet.create({
  container: {
    marginTop: "15rem",
  },
  header: {
    ...FONTS.h3,
    marginBottom: "10rem",
    alignSelf: "center",
  },
  listItem: {
    flexDirection: "row",
    alignItems: "center",
    paddingVertical: "15rem",
  },
  listText: {
    ...FONTS.h3,
    flex: 1,
    marginLeft: "10rem",
  },
});

export default PopableInput;
