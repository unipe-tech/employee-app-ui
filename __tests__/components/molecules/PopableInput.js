import { render, fireEvent, waitFor } from "@testing-library/react-native";
import { describe, expect, test } from "@jest/globals";
import PopableInput from "../../../components/molecules/PopableInput";

describe("PopableInput", () => {
  test("renders UI correctly", () => {
    const tree = render(<PopableInput />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
