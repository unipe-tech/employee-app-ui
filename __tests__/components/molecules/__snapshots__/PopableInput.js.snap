// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopableInput renders UI correctly 1`] = `
<View
  style={
    Object {
      "marginTop": 29.605263157894736,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": "#aaaaaa",
        "borderRadius": 5,
        "borderWidth": 1.5,
        "flexDirection": "row",
        "height": 50,
        "paddingHorizontal": 19.736842105263158,
        "width": "100%",
      }
    }
  >
    <TextInput
      autoCompleteType="off"
      editable={true}
      keyboardType="default"
      onChangeText={[Function]}
      placeholderTextColor="#808080"
      style={
        Object {
          "color": "#000000",
          "flex": 1,
          "fontFamily": "Montserrat-Regular",
          "fontSize": 14,
          "lineHeight": 18,
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "position": "relative",
            "zIndex": 1,
          },
          undefined,
        ]
      }
    >
      <Modal
        hardwareAccelerated={true}
        onRequestClose={[Function]}
        supportedOrientations={
          Array [
            "portrait",
            "portrait-upside-down",
            "landscape",
            "landscape-left",
            "landscape-right",
          ]
        }
        transparent={true}
        visible={false}
      />
      <View
        onLayout={[Function]}
        pointerEvents="none"
        style={
          Array [
            Object {
              "overflow": "hidden",
              "width": 100,
            },
            Array [
              false,
              false,
              false,
              Object {
                "left": 0,
              },
              Object {
                "position": "absolute",
                "transform": Array [
                  Object {
                    "translateX": -0,
                  },
                  Object {
                    "translateY": -0,
                  },
                ],
              },
              undefined,
            ],
          ]
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "flexDirection": "row",
              "opacity": 0,
              "transform": Array [
                Object {
                  "translateX": -5,
                },
              ],
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "black",
                  "borderRadius": 4,
                  "flex": 1,
                  "overflow": "hidden",
                  "zIndex": 1,
                },
                false,
                false,
              ]
            }
          />
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onLayout={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Text />
      </View>
    </View>
  </View>
</View>
`;
